src/engine/source/kvdb/src/kvdb.cpp

```diff
@@ -1,561 +1,529 @@
-                            }
-                        }
-                        break;
-
-                    case ACTION_ON_CF::DELETE:
-                        {
-                            s = db->Delete(WriteOptions(), handle, Slice(key));
-                            if(s.ok()) {
-                                LOG(INFO) << "[" << __func__ << "]" << " key deleted OK {" << key << "} from CF name : " << column_family_name << std::endl;
-                            }
-                            else {
-                                LOG(ERROR) << "[" << __func__ << "]" << " couldn't delete value from CF, error: " << s.ToString() << std::endl;
-                                result = false;
-                            }
-                        }
-                        break;
-
-                    case ACTION_ON_CF::READ_WITHOUT_VALUE_COPY:
-                        {
-                            PinnableSlice pinnable_val;
-                            s = db->Get(ReadOptions(), handle, Slice(key), &pinnable_val);
-                            if(s.ok()) {
-                                value = pinnable_val.ToString();
-                                LOG(INFO) << "[" << __func__ << "]" << " value obtained OK {" << key << ","
-                                << value << "} from CF name : " << column_family_name << std::endl;
-                                pinnable_val.Reset();
-                            }
-                            else {
-                                LOG(ERROR) << "[" << __func__ << "]" << " couldn't insert value into CF without copy, error: " << s.ToString() << std::endl;
-                                result = false;
-                            }
-                        }
-                        break;
-
-                    default:
-                        LOG(ERROR) << "[" << __func__ << "]" << " this case shouldn't be reacheable" << std::endl;
-                        result = false;
-                        break;
-                    }
-
-                }
-                s = db->DestroyColumnFamilyHandle(handle);
-                if(!s.ok()) {
-                    LOG(WARNING) << "[" << __func__ << "]" << " couldn't delete column family handler, error: " << s.ToString() << std::endl;
-                }
-            }
-            handlers.clear();
-
-            s = db->Close();
-            if(!s.ok()) {
-                LOG(WARNING) << "[" << __func__ << "]" << " couldn't close db file, error: " << s.ToString() << std::endl;
-            }
+    if (m_state == State::Open)
+    {
+        rocksdb::Status s =
+            m_db->Delete(kOptions.write, cfh->second, rocksdb::Slice(key));
+        if (s.ok())
+        {
+            WAZUH_LOG_INFO("key deleted OK [{}]", key);
+            return true;
         }
-        else {
-            LOG(ERROR) << "[" << __func__ << "]" << " couldn't open db file, error: " << s.ToString() << std::endl;
-            result = false;
+        else
```

Comment:
avoid the else's in arrow pattern by just flipping the condition and doing an early return

Reply:
right, done!